### PR TITLE
Implement autosave for Cloud projects

### PR DESCRIPTION
**Pull request by AlexandreSi**
This PR uses the Cache API implemented by browsers in order to have the autosave feature enabled for cloud projects.

Cache API seems to be available everywhere: https://developer.mozilla.org/en-US/docs/Web/API/Cache

### CODE REVIEW

File: newIDE/app/src/Profile/AuthenticatedUserProvider.js [hunk 1]
```diff
@@ -636,6 +637,7 @@ export default class AuthenticatedUserProvider extends React.Component<
     }
     this._markAuthenticatedUserAsLoggedOut();
     clearCloudProjectCookies();
+    burstCloudProjectAutoSaveCache();
```
Comment (ClementPasteau) [inline newIDE/app/src/Profile/AuthenticatedUserProvider.js:640]: can you add this in the doDeleteAccount() function?
(maybe worth creating one single function onLogout() or something similar

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectOpener.js [hunk 2]
```diff
@@ -31,6 +36,30 @@ export const generateOnOpen = (authenticatedUser: AuthenticatedUser) => async (
 |}> => {
   const cloudProjectId = fileMetadata.fileIdentifier;
 
+  if (cloudProjectId.startsWith(CLOUD_PROJECT_AUTOSAVE_PREFIX)) {
+    if (!('caches' in window)) throw new Error('Cache API is not available.');
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectOpener.js:41]: you can reuse your isCacheApiAvailable here

File: newIDE/app/src/Profile/AuthenticatedUserProvider.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/Profile/AuthenticatedUserProvider.js]: Good idea! I missed that

**Review by ClementPasteau — APPROVED** [review]
nice!

**Pull request merged**